internal/directives/git_pr_opener.go

```diff
@@ -298,15 +367,38 @@ func mustOpenPR(
 		},
 	)
 	if err != nil {
-		return false, fmt.Errorf("error listing pull requests: %w", err)
+		return nil, fmt.Errorf("error listing pull requests: %w", err)
 	}
 	if len(prs) == 0 {
-		return true, nil
+		return nil, nil
 	}
+	// If promotion names are incorporated into PR source branches, it's highly
+	// unlikely that we would have found more than one PR matching the search
+	// criteria. Accounting for the possibility of users specifying their own
+	// source branch names using an expression, there is somewhat more of a
+	// possibility of multiple PRs being found. We sort descending by creation
```

Comment:
Is this true? As far as I know, on GitHub, you can only have one open pull request (which we are querying for) for a target/source combination at a time. This makes sense because they otherwise would be literal duplicates from the perspective of the Git patch.

Reply:
You're right that only two identical PRs can be _open_. But I tested and you can open one, close it, and then open a new one that's identical.

Reply:
Yes, but the default behavior of `ListPullRequests` is to only query for _open_ pull requests, which makes me think that the result will always equal one?

Reply:
> but the default behavior of ListPullRequests is to only query for open pull requests

Ah... I'm glad you called this out.

I want to find open _and_ closed PRs.

Reply:
In that case, wouldn't it be better to first sort by state (open > closed) and then by date (in case all are closed)?

Reply:
🤷‍♂️ Maybe? There are _so many_ edge cases lurking here.

You could open PR 1, close it, open an identical 2, close it, then re-open 1.

With my sort, we'd incorrectly get 2 as the top result when 1 is the one we ought to be tracking. Mine cannot contend with this edge case.

By your suggestion, we'd get 1 as the top result while it remains open, but once it's closed/merged, the same sort would give us 2 as the top result.

Maybe the right order is open > merged > closed?

Reply:
Sort order is fixed specifically in 19f3e7d44c8c7967088aaba09361ced0bd84b88d